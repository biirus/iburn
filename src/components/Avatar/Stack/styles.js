export default ({ spacing, colors }) => {
  return {
    /* Styles applied to root element */
    root: {
      display: "flex",
      lineHeight: 1,
    },

    /* Styles applied to the wrapper of `Avatar` element */
    avatarWrapper: {
      position: "relative",
    },

    /* Styles applied to each `Avatar` element */
    avatar: {
      border: `2px solid ${colors.n0}`,
    },
  };
};
