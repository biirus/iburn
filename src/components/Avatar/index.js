import React from "react";
import cn from "classnames";
import PropTypes from "prop-types";

// enhancers
import { enhance } from "./enhnace";

/**
 * The basic Avatar component that renders it's content or image
 * by passing `src` or `srcSet` prop
 */
class Avatar extends React.Component {
  static defaultProps = {};

  static propTypes = {
    /**
     * Class name string to be merged to root node
     */
    className: PropTypes.string,
    /**
     * [JSS](http://cssinjs.org/react-jss/?v=v8.5.1) classes object notation
     */
    classes: PropTypes.object.isRequired,
    /**
     * The `srcSet` attribute for the `img` element.
     */
    srcSet: PropTypes.string,
    /**
     * The `sizes` attribute for the `img` element.
     */
    sizes: PropTypes.string,
    /**
     * The `src` attribute for the `img` element.
     */
    src: PropTypes.string,
    /**
     * Used in combination with `src` or `srcSet` to
     * provide an alt attribute for the rendered `img` element.
     */
    alt: PropTypes.string,
    /**
     * Used to render icon or text elements inside the Avatar.
     * `src` and `alt` props will not be used and no `img` will
     * be rendered by default.
     *
     * This can be an element, or just a string.
     */
    children: PropTypes.node,
  };

  clearProps = () => {
    const {
      classes,
      className,
      children,
      src,
      srcSet,
      sizes,
      alt,
      sheet,
      theme,
      ...rest
    } = this.props;

    return rest;
  };

  render () {
    const {
      classes,
      className,
      children,
      src,
      srcSet,
      sizes,
      alt,
    } = this.props;
    const classNames = cn(classes.root, className);

    return (
      <div {...this.clearProps()} className={classNames}>
        {src || srcSet ? (
          <img
            className={classes.img}
            alt={alt}
            src={src}
            srcSet={srcSet}
            sizes={sizes}
          />
        ) : (
          children
        )}
      </div>
    );
  }
}

export default enhance(Avatar);
