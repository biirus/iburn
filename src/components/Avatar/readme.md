### Image avatars

Image avatars can be created by passing standard img props `src` or `srcSet` into the component.

```js
<Avatar src="http://i.pravatar.cc/100" alt="John Doe" />
<Avatar src="http://i.pravatar.cc/200" style={{width: 80, height: 80}} alt="John Doe" />
<Avatar src="http://i.pravatar.cc/300" style={{width: 100, height: 100}} alt="John Doe" />
<Avatar
  srcSet="http://i.pravatar.cc/100, http://i.pravatar.cc/200 2x, http://i.pravatar.cc/300 3x"
  alt="John Doe"
/>
```

### Icon avatars

Icon avatars are created by passing an icon as `children`.

```js
<Avatar><Icon label="atlassian" /></Avatar>
<Avatar style={{backgroundColor: "tomato", color: "#333"}}><Icon label="lock" /></Avatar>
<Avatar><Icon label="backlog" /></Avatar>
```

### Letter avatars

Avatars containing simple characters can be created by passing your string as `children`.

```js
<Avatar>IB</Avatar>
<Avatar>AB</Avatar>
```
