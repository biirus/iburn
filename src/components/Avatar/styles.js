export default ({ colors, typography, spacing, }) => {
  const size = spacing.unit * 6;

  return {
    root: {
      ...typography.h600,
      color: colors.w700a,

      display: "inline-flex",
      alignItems: "center",
      justifyContent: "center",
      alignSelf: "baseline",
      overflow: "hidden",
      width: size,
      height: size,

      backgroundColor: colors.n100,
      borderRadius: "50%",
    },

    img: {
      width: "100%",
      height: "100%",
      objectFit: "cover",
    },
  };
};
