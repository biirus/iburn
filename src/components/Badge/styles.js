export default ({ spacing, palette, colors, borderRadius }) => {
  return {
    /* Styles applied to the root element */
    root: {
      display: "inline-flex",
      padding: [[spacing.small / 2, spacing.small]],
      lineHeight: 1,
      borderRadius,
    },

    /* Styles applied to the root element if `color="default"` */
    default: {
      backgroundColor: colors.n20a,
      color: colors.n400,
    },

    /* Styles applied to the root element if `color="primary"` */
    primary: {
      backgroundColor: palette.primary,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="error"` */
    error: {
      backgroundColor: palette.error,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="success"` */
    success: {
      backgroundColor: palette.success,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="warning"` */
    warning: {
      backgroundColor: palette.warning,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="info"` */
    info: {
      backgroundColor: palette.info,
      color: palette.text.invert,
    },
  };
};
