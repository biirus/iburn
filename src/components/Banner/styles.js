export default ({
  spacing,
  palette,
  shadows,
  typography,
  zIndex,
  transitions,
}) => {
  const iconSize = spacing.unit * 3;

  return {
    root: {
      ...typography.h100,

      position: "fixed",
      top: 0,
      left: 0,
      width: "100%",
      zIndex: zIndex.flag,

      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      padding: spacing.medium,

      boxShadow: shadows.small,
      backgroundColor: palette.backgrounds.default,
      color: palette.text.primary,

      opacity: 0,
      transform: `translateY(-100%)`,

      transition: transitions.create(["opacity", "transform",], "leavingScreen"),
    },

    open: {
      opacity: 1,
      transform: `translateY(0)`,

      transition: transitions.create(
        ["opacity", "transform",],
        "enteringScreen"
      ),
    },

    warning: {
      backgroundColor: palette.messages.warning,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.warning,
      },
    },

    error: {
      backgroundColor: palette.messages.error,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.error,
      },
    },

    info: {
      backgroundColor: palette.messages.info,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.info,
      },
    },

    success: {
      backgroundColor: palette.messages.success,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.success,
      },
    },

    icon: {
      flex: `0 0 ${iconSize}px`,
      lineHeight: 0,
    },

    label: {
      flex: `0 1 auto`,
      marginLeft: spacing.small / 2,
      overflow: "hidden",
      textOverflow: "ellipsis",
      whiteSpace: "nowrap",
      lineHeight: `${iconSize}px`,
    },
  };
};
