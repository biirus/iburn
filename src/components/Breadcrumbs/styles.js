export default theme => {
  return {
    /* Styles applied to the root element. */
    root: {
      overflow: "hidden",
      whiteSpace: "nowrap",
      textOverflow: "ellipsis",
      color: theme.palette.text.secondary,
    },

    /* Styles applied to the item's `span` element */
    item: {
      "&+$item::before": {
        content: `"/"`,
        display: "inline-block",
        width: theme.spacing.medium,
        textAlign: "center",
      },
    },

    /* Styles applied to the item's `a` element */
    link: {
      color: theme.palette.text.secondary,
      textDecoration: "none",

      "&:hover": {
        textDecoration: "underline",
      },
    },
  };
};
