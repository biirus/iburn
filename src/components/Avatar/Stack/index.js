import React from "react";
import cn from "classnames";
import PropTypes from "prop-types";

// enhancers
import withStyles from "react-jss";

// styles
import styles from "./styles";

/**
 * The component to render an array of `<Avatar />` in a row
 */
class AvatarStack extends React.Component {
  static propTypes = {
    /**
     * Class name string to be merged to root node
     */
    className: PropTypes.string,
    /**
     * [JSS](http://cssinjs.org/react-jss/?v=v8.5.1) classes object notation
     */
    classes: PropTypes.object.isRequired,
    /**
     * Array of `<Avatar />` components to be shown as stack
     */
    children: PropTypes.arrayOf(PropTypes.element).isRequired,
  };

  render () {
    const { className, classes, children, sheet, theme, ...rest } = this.props;
    const classNames = cn(classes.root, className);

    return (
      <div {...rest} className={classNames}>
        {React.Children.map(children, (child, index) => {
          return (
            <div
              key={index}
              className={classes.item}
              style={{ zIndex: children.length - index, }}
            >
              {React.cloneElement(child, { className: classes.inner, })}
            </div>
          );
        })}
      </div>
    );
  }
}

const enhance = withStyles(styles);

export default enhance(AvatarStack);
