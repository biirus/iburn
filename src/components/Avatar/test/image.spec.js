import React from "react";

import Avatar from "../index";
import styles from "../styles";

const classes = getClasses(styles);
const Component = Avatar.InnerComponent;

describe("<Avatar />", () => {
  describe("Image avatar", () => {
    it("should render a div containing an img", () => {
      const wrapper = shallow(<Component classes={classes} src="img.jpg" />);

      expect(wrapper.name()).toEqual("div");
      expect(wrapper.childAt(0).is("img")).toBeTruthy();
    });

    it("should pass img props to <img />", () => {
      const wrapper = shallow(
        <Component
          classes={classes}
          alt="John Doe"
          srcSet="john.png 1x, jane.png 2x"
        />
      );

      expect(wrapper.find("img").prop("alt")).toEqual("John Doe");
      expect(wrapper.find("img").prop("srcSet")).toEqual(
        "john.png 1x, jane.png 2x"
      );
    });

    it("should render img avatar correctly", () => {
      const wrapper = shallow(<Component classes={classes} src="img.jpg" />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
