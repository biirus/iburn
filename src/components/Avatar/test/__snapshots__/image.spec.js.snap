// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> Image avatar should render img avatar correctly 1`] = `
<div
  className="root"
>
  <img
    className="img"
    src="img.jpg"
  />
</div>
`;
